### ISSUE TITLE
MultiValueBitmapIndexTest.testConsecutiveQueries and MultiValueBitmapIndexTest.testRandomQueries

### ISSUE THREAD

**Issue by puzpuzpuz** [hazelcast/hazelcast#16544]
Got these failures here: http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/5633/

Stacktraces:
MultiValueBitmapIndexTest.testRandomQueries[IndexConfig{name=null, type=BITMAP, attributes=[habits[any]], bitmapIndexOptions=BitmapIndexOptions{uniqueKey=__key, uniqueKeyTransformation=RAW}}]
```
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.map.impl.query.MultiValueBitmapIndexTest.verifyQueries(MultiValueBitmapIndexTest.java:244)
	at com.hazelcast.map.impl.query.MultiValueBitmapIndexTest.testRandomQueries(MultiValueBitmapIndexTest.java:181)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

MultiValueBitmapIndexTest.testConsecutiveQueries[IndexConfig{name=null, type=BITMAP, attributes=[habits[any]], bitmapIndexOptions=BitmapIndexOptions{uniqueKey=__key, uniqueKeyTransformation=LONG}}]
```
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.map.impl.query.MultiValueBitmapIndexTest.verifyQueries(MultiValueBitmapIndexTest.java:244)
	at com.hazelcast.map.impl.query.MultiValueBitmapIndexTest.testConsecutiveQueries(MultiValueBitmapIndexTest.java:151)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by taburet** [comment]
With high probability the root cause is the same as in https://github.com/hazelcast/hazelcast/issues/16457

**Comment by vojtechtoman** [comment]
@taburet I think you are right, I managed to reproduce the failure using the same trick as in https://github.com/hazelcast/hazelcast/issues/16457

**Comment by ahmetmircik** [comment]
Failed in http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/5689/testReport/com.hazelcast.map.impl.query/MultiValueBitmapIndexTest/testConsecutiveQueries_IndexConfig_name_null__type_BITMAP__attributes__habits_any____bitmapIndexOptions_BitmapIndexOptions_uniqueKey___key__uniqueKeyTransformation_LONG___/

```
Error Message
expected:<1> but was:<0>
Stacktrace
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.map.impl.query.MultiValueBitmapIndexTest.verifyQueries(MultiValueBitmapIndexTest.java:246)
	at com.hazelcast.map.impl.query.MultiValueBitmapIndexTest.testConsecutiveQueries(MultiValueBitmapIndexTest.java:151)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by sancar** [comment]
Failed here again 

http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/5768/testReport/junit/com.hazelcast.map.impl.query/MultiValueBitmapIndexTest/testRandomQueries_IndexConfig_name_null__type_BITMAP__attributes__habits_any____bitmapIndexOptions_BitmapIndexOptions_uniqueKey_stringId__uniqueKeyTransformation_OBJECT___/

```
Error Message
expected:<1> but was:<0>
Stacktrace
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.map.impl.query.MultiValueBitmapIndexTest.verifyQueries(MultiValueBitmapIndexTest.java:246)
	at com.hazelcast.map.impl.query.MultiValueBitmapIndexTest.testRandomQueries(MultiValueBitmapIndexTest.java:194)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by taburet** [comment]
The fix is now ported to 4.0.z as well. Closing.
